### PR TITLE
Fix issue DPTP-3486 Add "--hosted"

### PR DESCRIPTION
**Pull request by bear-redhat**
Add a new parameter to include the names of hosted clusters.

### CODE REVIEW

**Comment by bear-redhat** [comment]
/hold

The breaking changes are expected:
```
Untracked files:
  (use "git add <file>..." to include in what will be committed)
        clusters/build-clusters/01_cluster/common_except_hosted
        clusters/build-clusters/02_cluster/common_except_hosted
        clusters/build-clusters/build03/common_except_hosted
        clusters/build-clusters/build04/common_except_hosted
        clusters/build-clusters/build05/common_except_hosted
        clusters/build-clusters/multi01/common_except_hosted
```

I will commit the changes in `release` once this PR is merged.

File: cmd/cluster-init/main.go [hunk 1]
```diff
@@ -53,6 +57,7 @@ func parseOptions() (options, error) {
 	fs.StringVar(&o.githubLogin, "github-login", githubLogin, "The GitHub username to use. Set to "+githubLogin+" by default")
 	fs.StringVar(&o.assign, "assign", githubTeam, "The github username or group name to assign the created pull request to. Set to Test Platform by default")
 	fs.BoolVar(&o.useTokenFileInKubeconfig, "use-token-file-in-kubeconfig", true, "Set true if the token files are used in kubeconfigs. Set to true by default")
+	fs.Var(&o.hostedClusters, "hosted", "Specify the name of a hosted cluster. This flag can be set multiple times to specify multiple clusters. Hosted clusters will not have 'common_except_hosted' applied to them.")
```
[the remark this answers is not in the document]
Comment (bear-redhat) [reply cmd/cluster-init/main.go:60]: If I am not mistaken, we are changing the [script](https://github.com/openshift/release/blob/7852ff31d9eb9db2c361f89de1c0e20ca301f982/hack/validate-ci-build-clusters.sh#L19-L18) to add parameters like `--hosted build08`

File: cmd/cluster-init/main_test.go [hunk 2]
```diff
@@ -89,6 +89,23 @@ func TestValidateOptions(t *testing.T) {
...
+		{
+			name: "valid with hosted true",
+			options: options{
+				clusterName: "newcluster",
+				releaseRepo: testdata,
+				hosted:      true,
+			},
+		},
+		{
+			name: "update single cluster with hosted true",
```
Comment (hongkailiu) [inline cmd/cluster-init/main_test.go:101]:
```suggestion
			name: "update an existing cluster with hosted true",
```

**Review by hongkailiu — COMMENTED** [review]
The code looks good to me.
Could you please add `host` cases into the integration tests which should cover, adding a new one and updating existing ones?

**Comment by hongkailiu** [comment]
Another thing: Could you update the PR with b08 with the files generated by this PR?

**Comment by hongkailiu** [comment]
/lgtm

Please make ready the PR to fix the breaking before cancel the hold.

**Comment by hongkailiu** [comment]
/lgtm cancel

Where do you use the new folder?

**Comment by bear-redhat** [comment]
The break changes are unavoidable I think, since we need to create the new `common_except_hosted` directories in existing clusters.

**Comment by hongkailiu** [comment]
> Where do you use the new folder?

Used in the soft link under `build0*`.

**Pull request merged**